packages/google_sign_in/extension_google_sign_in_as_googleapis_auth/lib/extension_google_sign_in_as_googleapis_auth.dart

```diff
@@ -0,0 +1,36 @@
+// Copyright 2020 The Flutter Authors
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file or at
+// https://developers.google.com/open-source/licenses/bsd
+
+import 'package:meta/meta.dart';
+import 'package:google_sign_in/google_sign_in.dart';
+import 'package:googleapis_auth/auth.dart' as googleapis_auth;
+import 'package:http/http.dart' as http;
+
+/// Extension on [GoogleSignIn] that adds an `authenticatedClient` method.
+///
+/// This method can be used to retrieve an authenticated [googleapis_auth.AuthClient]
+/// client that can be used with the rest of the `googleapis` libraries.
+extension GoogleApisGoogleSignInAuth on GoogleSignIn {
+  /// Retrieve a `googleapis` authenticated client.
+  Future<googleapis_auth.AuthClient> authenticatedClient({
+    @visibleForTesting GoogleSignInAuthentication authentication,
+    @visibleForTesting List<String> scopes = const [],
```

Comment:
Should things that only visible for testing have `debug` in their names?

Reply:
Yeah, good idea. Let me rename these two.